src/libCli/Call.cpp

```diff
@@ -146,7 +147,31 @@ namespace cli
         // Prepare the RPC call:
         std::multimap<grpc::string, grpc::string> clientMetadata;
         std::string methodStr = "/" + serviceName + "/" + methodName;
-        grpc::testing::CliCall call(channel, methodStr, clientMetadata);
+
+        // Get deadline for RPC from input or use custom
+        std::optional<std::chrono::time_point<std::chrono::system_clock>> timeout;
+        std::chrono::time_point<std::chrono::system_clock> defaultTimeout = std::chrono::system_clock::now() + std::chrono::milliseconds(100);
+
+
+        bool setTimeout = (parseTree.findFirstChild("rpcTimeout") != "");
+
+        if(!setTimeout){
+            if(method->client_streaming() || method->server_streaming()){
+                // Pass no timeout
+                timeout = std::nullopt;
+            }else{
+                timeout = defaultTimeout;
+            }
+        }
+
+        if(setTimeout){
+            std::string customTimeout = parseTree.findFirstChild("rpcTimeoutInMs");
+            int customTimeoutMs = std::stoi(customTimeout);
```

Comment:
what happens ifstoi fails to parse? did you test this?

Reply:
Do you mean e.g. if someone tries to provide a string as custom timeout? If so, the parse fails.

Reply:
yes exactly, so I woule expect a functional test implemented testing this and a readable error to be printed